src/chainlink/factory/BaseChainlinkOracleFactory.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {LibClone} from "solady/utils/LibClone.sol";
+import {IChainlinkOracleFactory} from "../../interfaces/IChainlinkOracleFactory.sol";
+import {ChainlinkOracleImpl} from "../oracle/ChainlinkOracleImpl.sol";
+
+/// @title Base Chainlink Oracle Factory
+/// @author 0xSplits
+/// @notice Factory for creating Chainlink Oracles
+abstract contract BaseChainlinkOracleFactory is IChainlinkOracleFactory {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using LibClone for address;
+
+    /// -----------------------------------------------------------------------
+    /// functions - public & external
+    /// -----------------------------------------------------------------------
+
+    /// @inheritdoc IChainlinkOracleFactory
+    function createOracle(bytes calldata data_, bytes32 salt_) external returns (address) {
+        return _createOracle(data_, salt_);
```

Comment:
any reason to use a separate internal fn if we're only going to have 1 (~untyped) fn to create oracles from?

also do we think only offering the blob version of the creation function is good enough? what's the tradeoff in your mind?

Reply:
1. no particular reason
2. yeah I think it is enough, unless the data itself was something easily readable and modifiable I would consider adding a separate create flow plus